### ISSUE TITLE
gRPC 1.56.2 fails to build using CMake

### ISSUE THREAD

**Issue by sid1029** [grpc/grpc#33887]
### What version of gRPC and what language are you using?
grpc git tag 1.56.2. Language : C++

### What operating system (Linux, Windows,...) and version?
Linux

### What runtime / compiler are you using (e.g. python version or version of gcc)
Clang 11.0.0

### What did you do?
Please provide either 1) A unit test for reproducing the bug or 2) Specific steps for us to follow to reproduce the bug. If there’s not enough information to debug the problem, gRPC team may close the issue at their discretion. You’re welcome to re-open the issue once you have a reproduction.

Cloned grpc tag 1.56.2 and tried to build + install it using cmake using the shell script :

```
ABSL_INSTALL_PATH=`(realpath ../abseil/lib/cmake/absl)`
PROTOBUF_INSTALL_PATH=`(realpath ../protobuf/lib/cmake/protobuf)`
GRPC_INSTALL_PATH=`(realpath .)`
git clone -j 16 --recurse-submodules -b v1.56.2 --depth 1 --shallow-submodules https://github.com/grpc/grpc grpc_src
cd grpc_src
rm -rf cmake/build
mkdir -p cmake/build
pushd cmake/build
cmake -DgRPC_INSTALL=ON \
      -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
      -DCMAKE_CXX_STANDARD=20 \
      -DCMAKE_INSTALL_PREFIX=$GRPC_INSTALL_PATH \
      -DCMAKE_BUILD_TYPE=Release \
      -DgRPC_ABSL_PROVIDER=package \
      -Dabsl_DIR=$ABSL_INSTALL_PATH \
      -DgRPC_PROTOBUF_PROVIDER=package \
      -DProtobuf_DIR=$PROTOBUF_INSTALL_PATH \
      -DgRPC_SSL_PROVIDER=package \
      -DgRPC_ZLIB_PROVIDER=package \
      -DgRPC_BUILD_TESTS=OFF \
      -DgRPC_BUILD_CODEGEN=ON \
      -DgRPC_BUILD_CSHARP_EXT=OFF \
      -DgRPC_BACKWARDS_COMPATIBILITY_MODE=OFF \
      -DgRPC_BUILD_GRPC_CPP_PLUGIN=ON \
      -DgRPC_BUILD_GRPC_CSHARP_PLUGIN=OFF \
      -DgRPC_BUILD_GRPC_NODE_PLUGIN=OFF \
      -DgRPC_BUILD_GRPC_OBJECTIVE_C_PLUGIN=OFF \
      -DgRPC_BUILD_GRPC_PHP_PLUGIN=OFF \
      -DgRPC_BUILD_GRPC_PYTHON_PLUGIN=ON \
      -DgRPC_BUILD_GRPC_RUBY_PLUGIN=OFF \
      ../..
cmake --build . --target install -j 64
```

### What did you expect to see?
gRPC builds and installs successfully.

### What did you see instead?
The CMake project generation succeeds. I verified it's able to find abseil and protobuf installations by printing out absl_VERSION and Protobuf_VERSION in cmake/abseil-cpp.cmake and cmake/protobuf.cmake respectively.
i.e. find_package(absl CONFIG) and find_package(Protobuf CONFIG) calls are successful.

During the build stage it fails to build the `grpc_plugin_support` target with errors like :

```
include/grpcpp/impl/codegen/config_protobuf.h:74:10: fatal error: 'absl/status/status.h' file not found
#include "absl/status/status.h"
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/grpc_plugin_support.dir/build.make:132: CMakeFiles/grpc_plugin_support.dir/src/compiler/php_generator.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:645: CMakeFiles/grpc_plugin_support.dir/all] Error 2
```

Looking at the generated cmake/build/CMakeFiles/grpc_plugin_support.dir/flags.make the CXX_INCLUDES flag is missing the abseil includes path.

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

### Anything else we should know about your project / environment?
Linux Environment :

Linux 5.4.0-150-generic #167-Ubuntu SMP Mon May 15 17:35:05 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

**Comment by sid1029** [comment]
The abseil installation is built from source at version : https://github.com/abseil/abseil-cpp/releases/tag/20230125.3
The protobuf installation is built from source at version : https://github.com/protocolbuffers/protobuf/releases/tag/v21.8

**Comment by mialuyao** [comment]
I got the problem as 

> CMake Error at /root/library-tools/grpc/examples/cpp/cmake/common.cmake:99 (find_package):
  Found package configuration file:
    /usr/local/lib/cmake/protobuf/protobuf-config.cmake
  but it set Protobuf_FOUND to FALSE so package "Protobuf" is considered to
  be NOT FOUND.  Reason given by package:
  The following imported targets are referenced, but are missing:
  absl::absl_check absl::absl_log absl::die_if_null absl::log_initialize
Call Stack (most recent call first):
 CMakeLists.txt:24 (include)

But all absl files have been `git submodule init.`

**Comment by yashykt** [comment]
I'm not sure..
Are you able to build protobuf successfully?

**Comment by veblush** [comment]
Not sure what happened to your case but you can see https://github.com/grpc/grpc/blob/master/test/distrib/cpp/run_distrib_test_cmake.sh script which builds gRPC with cmake.

**Comment by dazfez** [comment]
I'm getting this issue when trying to build gRPC 1.57.0 using Conan.
This is the error that I'm getting even though the abseil dependency has been installed:-

`C:\Users\ContainerAdministrator\.conan2\p\b\grpc615fb20d017b4\b\src\include\grpcpp/impl/codegen/config_protobuf.h(74): fatal error C1083: Cannot open include file: 'absl/status/status.h': No such file or directory`

Building version 1.54.3 works successfully though.

I'm not a C++ developer but it looks like an issue in [https://github.com/grpc/grpc/commits/master/include/grpcpp/impl/codegen/config_protobuf.h](https://github.com/grpc/grpc/commits/master/include/grpcpp/impl/codegen/config_protobuf.h)

where the reference to Status was changed in this commit:- [https://github.com/grpc/grpc/commit/0f1afec5a8906ae3f00d58ff2499ad7e7cfb577e#diff-18e80b33dc8068b50969735abc5f676ba067d83eef33bfc7ac411f47020e7fba](url)

**Comment by bensontruong** [comment]
This issue keeps getting closed by the bot. I'm seeing the same issue building with cmake. Dependencies are being built outside of the thirdparty directories. What other information needs to be provided?

**Comment by bhutch29** [comment]
I'm having this trouble building 1.65.0 (latest version on conan-center). Did anyone ever resolve the issue?

**Comment by joda01** [comment]
Hello @bhutch29 I have the same problem. Did you found a solution right now?

**Comment by JesseLovelace** [comment]
For anyone still experiencing this issue, I'm pretty sure I understand what the problem is. It's a problem with the CmakeLists.txt file in this version of grpc. The `grpc_plugin_support` target requires `absl::status`, but [does not link it](https://github.com/grpc/grpc/blob/v1.56.2/CMakeLists.txt#L4701). This was fixed in [this commit](https://github.com/grpc/grpc/commit/da58cffb04c78cfba42fe0783e6bf3e7068513e1#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR6092) which was released in 1.69.0

Upgrading to 1.69.0 or later should fix this issue
